drivers/hmis/app/models/hmis/ce/match/expression/custom_assessment_field_map.rb

```diff
@@ -0,0 +1,129 @@
+# frozen_string_literal: true
+
+module Hmis::Ce::Match::Expression
+  class CustomAssessmentFieldMap
+    include Memery
+
+    def initialize(current_date: Date.current)
+      @current_date = current_date
+    end
+
+    def client_query(clients, field)
+      form_definition_identifier, field_name = parse_entity_type(field)
+      column = column_for_field(field_name)
+      client_ids = clients.pluck(:id)
+
+      result = Hmis::DestinationClientLatestAssessment.
+        where(destination_client_id: client_ids).
+        where(form_identifier: form_definition_identifier).
+        joins(:custom_assessment).
+        pluck(:destination_client_id, column).
+        to_h
+
+      # Ensure all clients are in the hash, setting a default value for those missing.
+      client_ids.each do |client_id|
+        # we might need to support booleans in the future
+        result[client_id] = nil unless result.key?(client_id)
+      end
+
+      result
+    end
+
+    def joins(_field)
+      # Join latest-assessment view and the associated custom assessment
+      [{ destination_client_latest_assessments: :custom_assessment }]
+    end
+
+    def arel_field(field)
```

Comment:
Note, we can apply this approach to the CDE field map too. I tried it and it does work for pre-filtering. However I'm a little nervous about it because of the potential row-explosion with the join to CDEs